BinaryEncoding.md

```diff
@@ -390,10 +387,10 @@ out of range, `br_table` branches to the default target.
 
 The `memory_immediate` type is encoded as follows:
 
-| Name | Type | Present? | Description |
-| ---- | ---- | ---- | ---- |
-| flags | `uint8` | always | a bitfield where<br>bit `4` indicates an offset follows<br>bit `7` indicates natural alignment<br>other bits are reserved for future use |
-| offset | `varuint32` | `flags[4] == 1` | the value of the offset |
+| Name | Type | Description |
+| ---- | ---- | ---- |
+| offset | `varuint32` | the value of the offset |
+| alignment | `varuint32` | the value of the alignment |
```

Comment:
Please also see the suggestion in https://github.com/WebAssembly/design/pull/571 which optimizes for the common case of natural alignment and for the  power-of-two values and the expected maximum alignment of the page size. One key property is that the flag is zero for natural alignment and does not change - whereas if the actual alignment is encoded then it will change based on the natural alignment. Encoding the powers of two keeps the values smaller, and if small values are move frequent then this might help compression. A custom compressor could easily map the proposed leb128 values to short codes, but if depending on a generic compressor then it should help to map the frequent case to a consistent code and map the values to high frequency values. Also keep in mind that Brotli only looks back at most two bytes so packing can still help.